File: Algorithms/Pattern Searching/Naive Method/naive-search.cpp
/*
 * Naive Pattern Searching
 *
 */

#include <iostream>
#include <cstring>

using namespace std;

// Time complexity: O(m*(n-m+1))
void search(char *pattern, char *text) {
    int M = strlen(pattern);
    int N = strlen(text);

    // A loop to slide pattern[] one by one
    for (int i = 0; i <= N - M; i++) {
        int j;

        // For current index i, check for patternt match
        for (j = 0; j < M; j++)
            if (text[i+j] != pattern[j])
                break;

        // if pattern[0...M-1] = text[i, i+1, ...i+M-1]
        if (j == M)
           cout<<"pattern found at index n = "<<i<<endl;
    }
}

int main() {
   char text[] = "AABAACAADAABAAABAA";
   char pattern[] = "AABA";
   search(pattern, text);
   return 0;
}

File: Algorithms/Pattern Searching/Naive Method/naive-search.py
"""
    Naive Pattern Searching
"""


def search(pattern, txt):
    M = len(pattern)
    N = len(txt)

    # A loop to slide pattern[] one by one
    for i in range(N-M+1):
        for j in range(M):
            if txt[i+j] != pattern[j]:
                break
        if j == M-1:  # if pattern[0...M-1] = txt[i, i+1, ...i+M-1]
            print("Pattern found at index " + str(i))


txt = "AABAACAADAABAAABAA"
pattern = "AABA"
search(pattern, txt)
